### PR TITLE
implement heap and priority_queue

### PR DESCRIPTION
**Pull request by levyfan**
implement heap and priority_queue as suggested in https://github.com/RedisLabsModules/redex/pull/3

### CODE REVIEW

File: rmutil/heap.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+void Make_Heap(Vector *v, size_t first, size_t last, int (*cmp)(void *, void *));
+
+
+/* Push element into heap range
+ * Given a heap in the range [first,last-1), this function extends the range considered a heap to [first,last) by
+ * placing the value in (last-1) into its corresponding location within it.
+ * A range can be organized into a heap by calling make_heap. After that, its heap properties are preserved if elements
+ * are added and removed from it using push_heap and pop_heap, respectively.
+ */
+void Push_Heap(Vector *v, size_t first, size_t last, int (*cmp)(void *, void *));
```
Comment (dvirsky) [inline rmutil/heap.h]: Rename to Heap_Push

File: rmutil/heap.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+/* Pop element from heap range
+ * Rearranges the elements in the heap range [first,last) in such a way that the part considered a heap is shortened
+ * by one: The element with the highest value is moved to (last-1).
+ * While the element with the highest value is moved from first to (last-1) (which now is out of the heap), the other
+ * elements are reorganized in such a way that the range [first,last-1) preserves the properties of a heap.
+ * A range can be organized into a heap by calling make_heap. After that, its heap properties are preserved if elements
+ * are added and removed from it using push_heap and pop_heap, respectively.
+ */
+void Pop_Heap(Vector *v, size_t first, size_t last, int (*cmp)(void *, void *));
```
Comment (dvirsky) [inline rmutil/heap.h]: Rename to Heap_Pop

**Comment by dvirsky** [comment]
Very nice. Just a couple of naming convention fixes, and we're good. Thank you very very much!

**Comment by levyfan** [comment]
Thanks for reviewing! I fix the naming and squash my commits into one.

**Comment by levyfan** [comment]
@dvirsky @itamarhaber please have a review.

**Comment by dvirsky** [comment]
looks good! thanks.

**Pull request merged**